pkg/drain/drain.go

```diff
@@ -64,13 +45,33 @@ func GetPodsForDeletion(kc kubernetes.Interface, node string) (pods []v1.Pod, er
 	return pods, nil
 }
 
-// Pared down version of
-// https://github.com/kubernetes/kubernetes/blob/cbbf22a7d2b06a55066b16885a4baaf4ce92d3a4/pkg/kubectl/cmd/drain.go's
-// getDaemonsetController().
-func getDaemonsetController(kc kubernetes.Interface, sr *v1.SerializedReference) (interface{}, error) {
-	switch sr.Reference.Kind {
+// getOwnerDaemonset returns true if a pod has any existing DaemonSet owner.
+func getOwnerDaemonset(kc kubernetes.Interface, pod v1.Pod) (interface{}, error) {
+	if len(pod.OwnerReferences) == 0 {
+		return nil, fmt.Errorf("pod %q has no owner objects", pod.Name)
+	}
+	for _, ownerRef := range pod.OwnerReferences {
+		// skip pod if it is owned by an existing daemonset
+		if ownerRef.Kind == "DaemonSet" {
+			ds, err := getDaemonsetController(kc, pod.Namespace, &ownerRef)
+			if err == nil {
+				// daemonset owner exists
+				return ds, nil
+			}
+			if !errors.IsNotFound(err) {
+				return nil, fmt.Errorf("failed to get controller of pod %q: %v", pod.Name, err)
+			}
+		}
+	}
+	// pod may have owners, but they don't exist or aren't daemonsets
+	return nil, fmt.Errorf("pod %q has no existing damonset owner", pod.Name)
+}
+
+// Stripped down version of https://github.com/kubernetes/kubernetes/blob/1bc56825a2dff06f29663a024ee339c25e6e6280/pkg/kubectl/cmd/drain.go#L272
+func getDaemonsetController(kc kubernetes.Interface, namespace string, controllerRef *v1meta.OwnerReference) (interface{}, error) {
+	switch controllerRef.Kind {
 	case "DaemonSet":
-		return kc.ExtensionsV1beta1().DaemonSets(sr.Reference.Namespace).Get(sr.Reference.Name, v1meta.GetOptions{})
+		return kc.Extensions().DaemonSets(namespace).Get(controllerRef.Name, v1meta.GetOptions{})
```

Comment:
Even if drain uses Extensions, I think the versioned client is probably correct.